packages/cli/src/ceramic-daemon.ts

```diff
@@ -511,51 +569,107 @@ export class CeramicDaemon {
+    const result = await this.ceramic.did.verifyJWS(jws)
+    return {
+      kid: result.kid,
+      code: result.payload.code,
+      requestPath: result.payload.requestPath,
+      models: result.payload.requestBody ? result.payload.requestBody.models : undefined,
+    }
   }
 
-  private  _validateModelIDStrings(
-    modelIDStrings: any
-  ): {
-    modelIDStrings?: Array<string>,
-    error?: string
-  } {
-    const cast = modelIDStrings as Array<string>
-    let error = undefined
-    if (!cast || cast.length === 0) {
-      error = 'The `models` parameter is required and it has to be an array containing at least one model stream id'
-    }
+  private async _validateAdminApiJWS(
+    basePath: string,
+    jws: string | undefined,
+    shouldContainModels: boolean
+  ): Promise<AdminApiJWSValidationResult> {
+    if (!jws) return { error: `Missing authorization jws` }
 
-    return {
-      modelIDStrings: error ?? cast,
-      error: error,
+    let parsedJWS
+    try {
+      parsedJWS = await this._parseAdminApiJWS(jws)
+    } catch (e) {
+      return { error: `Error while processing the authorization header ${e.message}` }
+    }
+    if (parsedJWS.requestPath !== basePath) {
+      return { error: `The jws block contains a request path that doesn't match the request`}
+    } else if (!parsedJWS.code) {
+      return { error: 'Admin code is missing from the the jws block' }
+    } else  if (shouldContainModels && (!parsedJWS.models || parsedJWS.models.length === 0)) {
+      return { error: `The 'models' parameter is required and it has to be an array containing at least one model stream id`}
+    } else {
+      return {
+        kid: parsedJWS.kid,
+        code: parsedJWS.code,
+        models: parsedJWS.models?.map( modelIDString => StreamID.fromString(modelIDString) )
+      }
     }
   }
 
-  async startIndexingModels(req: Request, res: Response): Promise<void> {
-    const { modelIDStrings, error } = this._validateModelIDStrings(req.body.models)
-    if (error) {
-      res.status(422).json({ error: error })
+  private async _processAdminModelsMutationRequest(
+    req: Request,
+    res: Response,
+    successCallback: AdminApiModelMutationMethod
+  ): Promise<void> {
+    const jwsValidation = await this._validateAdminApiJWS(
+      req.baseUrl,
+      req.body.jws,
+      true
+    )
+    if (jwsValidation.error) {
+      res.status(StatusCodes.UNPROCESSABLE_ENTITY).json({ error: jwsValidation.error })
     } else {
-      await this.ceramic.admin.startIndexingModels(modelIDStrings.map( modelIDString => StreamID.fromString(modelIDString) ))
-      res.status(200).json({ result: 'success' })
+      try {
+        this.verifyAndDiscardAdminCode(jwsValidation.code)
+        await successCallback(jwsValidation.kid, jwsValidation.models)
+        res.status(StatusCodes.OK).json({ result: 'success' })
+      } catch (e) {
+        res.status(StatusCodes.UNAUTHORIZED).json({ error: e.message })
```

Comment:
unauthorized is only the right error code if `verifyAndDiscardAdminCode` fails.  If successCallback fails it should be something else.